aten/src/ATen/native/cudnn/ConvShared.cpp

```diff
@@ -565,6 +555,23 @@ Tensor cudnn_convolution_add_relu(
       /*device=*/kCUDA,
       /*pin_memory=*/c10::nullopt,
       /*memory_format=*/at::MemoryFormat::Contiguous);
+  at::native::cudnn_convolution_relu_out(
+    input_t, weight_t, bias_t, stride, padding, dilation, groups, output_t);
+  return output_t;
+}
+
+Tensor& cudnn_convolution_add_relu_out(
+    const Tensor& input_t,
+    const Tensor& weight_t,
+    const Tensor& z_t,
+    const c10::optional<Scalar>& alpha,
+    const c10::optional<Tensor>& bias_t,
+    IntArrayRef stride,
+    IntArrayRef padding,
+    IntArrayRef dilation,
+    int64_t groups,
+    Tensor& output_t
+    ) {
   if (output_t.numel() == 0) {
```

Comment:
`cudnn_convolution_add_relu_out` is a public API, which means that users can send arbitrarily-sized `output_t` arguments to it. 0-sized `output_t` doesn't mean early exit, incorrectly-sized `output_t` has to either error out or be resized. We are trying to deprecate resizing of incorrectly sized `out` args, except resizing 0-element to correct size, so erroring out is better I'd say.

Reply:
@ngimel sounds good. suggestion for which error to use?

Reply:
You can use `at::native::resize_output` which will throw warning for now if output is of the wrong size, then at some point we'll switch it to error.

Reply:
@ngimel all set. if this is correct can you please approve.